src/lib.rs

```diff
@@ -65,124 +67,147 @@ pub fn program_from_file(bin_path: &str) -> Vec<U256> {
         let raw_opcode_u256 = U256::from_big_endian(&raw_opcode_bytes);
         program_code.push(raw_opcode_u256);
     }
-    program_code
+    Ok(program_code)
 }
 
 /// Run a vm program with a clean VM state.
-pub fn run_program(
+pub fn run_program(bin_path: &str, vm: VMState, tracers: &mut [Box<&mut dyn Tracer>]) -> Output {
+    match run_program_with_error(bin_path, vm, tracers) {
+        Ok((storage, vm)) => Output {
+            storage_zero: storage,
+            vm_state: vm,
+            reverted: false,
+            reason: None,
+        },
+        Err(e) => Output {
+            storage_zero: U256::zero(),
+            vm_state: VMState::default(),
+            reverted: true,
+            reason: Some(e),
+        },
+    }
```

Comment:
`Output` can not implement `Default`, as `Err` needs to be assinged a specific variable:
```rs
       ...
        Err(e) => Output {
            storage_zero: U256::zero(),
            vm_state: VMState::default(),
            reverted: true,
            reason: Some(e), //  <-- prohibits Default
        },
 ```
 Am I missing something?

Reply:
No, I missed that it was assigning the error 🤦

Reply:
That said, implementing `from` for the relevant types (`(U256, VMState)` and `EraVmError`) enables the following:
```rust
run_program_with_error(bin_path, vm, tracers)
    .map(Output::from)
    .ok_or_else(Output::from)
```
It may or may not require using the turbofish to specify types, but my guess is the compiler can infer them.